spring-security-modules/spring-security-web-boot-4/src/main/java/com/baeldung/apikeyauthentication/configuration/ApiKeyAuthentication.java

```diff
@@ -22,4 +22,14 @@ public Object getCredentials() {
     public Object getPrincipal() {
         return apiKey;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        return super.equals(obj);
```

Comment:
Is there a specific reason why we need to override these? And we are just returning what the superclass have so this seems unnecessary.

Reply:
I have a sonar lint notification that warns me to override also the equals and hashCode since the methods are overridden in the super class. We can remove it.